### PR TITLE
Fixes #3208 - print warnings for old enc/report routes

### PR DESCRIPTION
**Pull request by GregSutcliffe**
We should revert this once 1.3 is out...

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@
 Foreman::Application.routes.draw do
   #ENC requests goes here
   match "node/:name" => 'hosts#externalNodes', :constraints => { :name => /[^\.][\w\.-]+/ }
+  # To be removed for 1.4
+  match 'reports/create',     to: 'api/v2/reports#report_deprecation_msg', via: 'post'
```
Comment (domcleal) [inline config/routes.rb]: Looks like Ruby 1.9 (only) hash syntax?

**Comment by domcleal** [comment]
A reference to the upgrade notes would be great, don't worry about a link.  I'll merge this straight to 1.3-stable.

**Comment by GregSutcliffe** [comment]
@domcleal good catch, how's that?

**Comment by lzap** [comment]
Thanks! I have updated:

http://projects.theforeman.org/projects/foreman/wiki/Troubleshooting#After-Foreman-13-upgrade-reports-and-ENC-fact-uploads-no-longer-work

**Comment by domcleal** [comment]
@GregSutcliffe test failures

File: app/controllers/api/v2/hosts_controller.rb [hunk 2]
```diff
@@ -53,6 +53,13 @@ def facts
         render :json => {'message'=>e.to_s}, :status => :unprocessable_entity
       end
 
+      def enc_deprecation_msg
+        msg  = "/fact_values/create is deprecated, update your ENC to POST facts to /api/hosts/facts\n"
+        msg += '  See the Foreman 1.3 release notes for a new example ENC script'
+        logger.error "DEPRECATION: #{msg}."
+        render :json => {:message => msg}, :status => 400
```
Comment (domcleal) [inline app/controllers/api/v2/hosts_controller.rb:60]: These requests don't accept JSON, but probably YAML instead?

**Comment by domcleal** [comment]
I'm getting logging correctly, but the puppet agent and master don't log a single thing.  Any idea why?

**Comment by GregSutcliffe** [comment]
Nope, the master should log that it got a format it didn't like...

File: app/controllers/api/v2/hosts_controller.rb (hunk 2 above)
Comment (GregSutcliffe) [inline app/controllers/api/v2/hosts_controller.rb:60]: Does it matter? It's going to break Puppet anyway :)

**Comment by domcleal** [comment]
Merged as 7e5195b for Foreman 1.3.0-RC4 (only).  Thanks @GregSutcliffe.

**Pull request closed without merging**